models/settings_helpers_test.go

```diff
@@ -153,60 +153,70 @@ func TestSettings(t *testing.T) {
 			assert.EqualError(t, err, `data_retention: minimal resolution is 24h`)
 		})
 
-		t.Run("STT validation", func(t *testing.T) {
-			_, err := models.UpdateSettings(sqlDB, &models.ChangeSettingsParams{
-				EnableSTT:  true,
-				DisableSTT: true,
-			})
-			assert.EqualError(t, err, `Both enable_stt and disable_stt are present.`)
-
-			// Check we can enable STT while telemetry is enabled
+		t.Run("Telemetry and STT validation", func(t *testing.T) {
+			// ensure initial default state
 			ns, err := models.UpdateSettings(sqlDB, &models.ChangeSettingsParams{
-				EnableSTT: true,
+				EnableTelemetry: true,
+				DisableSTT:      true,
 			})
 			assert.NoError(t, err)
-			assert.True(t, ns.SaaS.STTEnabled)
+			assert.False(t, ns.Telemetry.Disabled)
+			assert.False(t, ns.SaaS.STTEnabled)
 
-			// Re-disable STT for the next tests
 			_, err = models.UpdateSettings(sqlDB, &models.ChangeSettingsParams{
-				DisableSTT: true,
+				EnableTelemetry:  true,
+				DisableTelemetry: true,
 			})
-			assert.NoError(t, err)
+			assert.EqualError(t, err, `Both enable_telemetry and disable_telemetry are present.`)
 
-			// Ensure telemetry is disabled
 			_, err = models.UpdateSettings(sqlDB, &models.ChangeSettingsParams{
-				DisableTelemetry: true,
+				EnableSTT:  true,
+				DisableSTT: true,
 			})
-			assert.NoError(t, err)
+			assert.EqualError(t, err, `Both enable_stt and disable_stt are present.`)
 
 			_, err = models.UpdateSettings(sqlDB, &models.ChangeSettingsParams{
-				EnableSTT: true,
+				EnableSTT:        true,
+				DisableTelemetry: true,
 			})
-			assert.EqualError(t, err, `cannot enable STT while telemetry is disabled`)
+			assert.EqualError(t, err, `Cannot enable STT while disabling telemetry.`)
 
-			_, err = models.UpdateSettings(sqlDB, &models.ChangeSettingsParams{
+			// enable both
```

Comment:
Maybe it's better to disable them before enabling? Because they are already enabled

Reply:
Nope, STT is disabled at this point